qemu/tests/virt_firmware_check_phys_bits.py

```
@@ -0,0 +1,96 @@
+       message in virt firmware log
+       limitation message: limit PhysBits to 46 (avoid 5-level paging)
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+    phys_bits_grep_cmd = params["phys_bits_grep_cmd"]
+    host_phys_bits = process.getoutput(phys_bits_grep_cmd, shell=True).strip()
+    if not host_phys_bits.isdigit():
+        test.error("Failed to get host phys-bits, the actual output is '%s'"
+                   % host_phys_bits)
+    host_phys_bits_limit = params["host_phys_bits_limit"]
+    params["cpu_model_flags"] %= host_phys_bits_limit
+    err_msg = params.get("err_msg")
+    ignored_err_msg = params.get("ignored_err_msg")
+    try:
+        error_context.context("Start the vm with host-phys-bits-limit=%s."
+                              % host_phys_bits_limit, test.log.info)
+        vm_name = params["main_vm"]
+        env_process.preprocess_vm(test, params, env, vm_name)
+        vm = env.get_vm(vm_name)
+        vm.verify_alive()
+    except virt_vm.VMCreateError as e:
+        if err_msg:
+            if err_msg not in str(e):
+                test.fail("Boot a vm with invalid phys-bits '%s', "
+                          "the error message is not the expected value "
+                          "'%s'. The actual output is '%s'."
+                          % (host_phys_bits_limit, err_msg, str(e)))
+        elif ignored_err_msg:
+            if not re.search(ignored_err_msg, str(e), re.S | re.I):
+                test.fail("Boot a vm with phys-bits '%s', the ignored "
+                          "error message is not the expected value "
+                          "'%s'. The actual output is '%s'."
+                          % (host_phys_bits_limit, ignored_err_msg, str(e)))
+        else:
+            test.error("Failed to create a vm, the error is '%s'" % str(e))
+    else:
```

I must lost something in my last comments...

in else clause, think about that, if we expect that the vm creation should fail, but we start the vm unexpectedly.
ie. we set err_msg or ignored_err_msg, but we never get an exception

---

As we discussed, added statement "if err_msg" after line 68 to handle vm started unexpectedly. Not added "if ignored_err_msg", since not all hosts have ignored_err_msg when setting phys-bits=36. I.E. It is acceptable for vm to start or not to start when setting phys-bits=36. Thanks.

---

make sense.